### DISCUSSION TITLE
Python Debug

### DISCUSSION THREAD

**Discussion by MrZmcH** [eclipse-theia/theia#13546] [Q&A]
I am using theia 1.44.0. How can I debug a python program when I know that the ms-python plugin does not work.

**Comment by msujew** [comment]
Hey @MrZmcH,

You mean [this plugin](https://open-vsx.org/extension/ms-python/debugpy)? It's supposed to work. If it doesn't, please open a bug report.

**Reply by MrZmcH** [reply]
@msujew [this issue](https://github.com/eclipse-theia/theia/issues/12219#issuecomment-1443535930),I had "Pylance Not Found" when installing ms-python in the same way.So I want to ask, if you can't install the ms-python plugin, is there any way to debug python?

**Reply by msujew** [reply]
You can install the linked extension independently of the whole `ms-python` plugin suite. The Pylance extension is not available on the open-vsx marketplace due to licensing issues.

**Reply by MrZmcH** [reply]
@msujew When installing python debug separately, the error "Select Python interpreter" appears, which means that "ms-python" is still needed. Do you know of any other extensions available?
![image](https://github.com/eclipse-theia/theia/assets/45864434/4cb15242-d266-4ad0-ad19-fe295bbbebdd)

**Reply by msujew** [reply]
Using the Pyright extension should work. It implements most of the features as the Pylance extension.

**Reply by MrZmcH** [reply]
@msujew I'm sorry, maybe I'm using it the wrong way, just adding "python-debug" and "pyright" these two extensions, still can't debug python programs, there is still "Select Interpreter" error message, I added "theiaPlugins" to package.json in the browser folder.

**Reply by msujew** [reply]
I see. I'll need to test this once I'm back from vacation in a few days.

**Reply by MrZmcH** [reply]
@msujew Excuse me, have you check this question?

**Comment by MrZmcH** [comment]
@msujew Hi,I wonder if you have time to read it recently, because I am still affected by this problem.

**Reply by msujew** [reply]
Alright, so I've just installed the `Python` extension suite (except for `Pylance`) and it just works out of the box for me:

![image](https://github.com/eclipse-theia/theia/assets/4377073/a2c52424-148d-4957-b62f-bf9fe6e64f3d)

**Reply by MrZmcH** [reply]
@msujew Have you installed other plugins? What is the version of theia? Can you share this example? thank you~

**Reply by msujew** [reply]
I used the [browser example app (current `master`)](https://github.com/eclipse-theia/theia/tree/master/examples/browser) and just installed the [python plugin suite](https://open-vsx.org/extension/ms-python/python). That was everything I did.

**Reply by MrZmcH** [reply]
@msujew Maybe I just need to ask one last question,How do you install the Python extension suite (except for Pylance)?

**Reply by msujew** [reply]
I installed it via the vsx-registry view widget.

**Reply by MrZmcH** [reply]
@msujew I want to add ms-python to theiaPlugins in package.json instead of installing it in vsx-registry. Is this supported?

**Reply by MrZmcH** [reply]
I tried to do this, but there was a pylance not found error when loading the plugin

**Reply by msujew** [reply]
Sorry for the late response. Yes, that's supported, you just need to exclude the extension. See [here](https://github.com/eclipse-theia/theia/blob/746f8ff1aeb0922dcb601dec7c9ca52ce0138118/package.json#L111-L120).

**Comment by MrZmcH** [comment]
@msujew Thank you for your reply. I tried it and there were still some problems, I was using theia version 1.48.0 and installed mspython version 2024.0.0, and there were some errors after starting theia. The following is the configuration of package.json
`"theiaPluginsDir": "plugins",
  "theiaPlugins": {
    "vscode-python": "https://open-vsx.org/api/ms-python/python/2024.0.0/file/ms-python.python-2024.0.0.vsix"
  },
  "theiaPluginsExcludeIds": [
    "ms-python.vscode-pylance"
  ]`
There are too many error messages.Maybe I'm using the wrong version.
![image](https://github.com/eclipse-theia/theia/assets/45864434/28e0cc91-24b6-437d-b14a-5d0939636c99)
![image](https://github.com/eclipse-theia/theia/assets/45864434/49e9d772-a4d2-4967-92a5-7a4f81dd9c4f)
